Fixed CPU Time not parsed properly when the TotalCPU column does not contain usec

jitkang opened the pull request:
The usec portion of the CPU Time is not properly parsed when the `TotalCPU` column from `sacct` command does not contain usec, for example `1:00:00` instead of `1:00:00.0`. Appending `.0` to the end of TotalCPU column will resolve this issue. This will also indirectly resolve CPU Efficiency calculation, as it depends on the value of CPU Time.

I am not sure if this problem apply to all the SLURM version, as we have only tested on version `21.08.8-2` in our cluster.

neilmunday commented:
Thanks for the the pull request @jitkang. I think a better way to handle the missing usec would be to adjust the regular expression used to parse the Slurm time string so that the `usec` group is optional like `hours`.

E.g.

```python
timeRe = re.compile(
    r"((?P<days>\d+)-)?((?P<hours>\d+):)?(?P<mins>\d+):(?P<secs>\d+)(\.(?P<usec>\d+))?"
)

...

if match.group("usec"):
    usec += int(match.group("mins")) * 1000000 * 60
```

jitkang commented:
Hey @neilmunday, I believe your suggestion to update the regular expression would be better in the long run. For me, I am just bad at writing regular expression code, so I opted for a simpler append of non-existing usec.

jitkang commented:
I have reverted the earlier changes, and updated the code with your suggested regular expression. I have run several test in our cluster, and it seems to be working fine.
